Декоративные элементы всех страниц

Alexander-Kirichenko opened the pull request:
---
:mortar_board: [Декоративные элементы всех страниц](https://up.htmlacademy.ru/htmlcss/36/user/2289265/tasks/16)

:boom: https://htmlacademy-htmlcss.github.io/2289265-sedona-36/7/

Alexander-Kirichenko committed:
Update style.css

Alexander-Kirichenko committed:
Update style.css

Alexander-Kirichenko committed:
Update style.css

Alexander-Kirichenko committed:
Update style.css

Alexander-Kirichenko committed:
Update index.html

Alexander-Kirichenko committed:
Update style.css

Alexander-Kirichenko committed:
делает view-plates

Alexander-Kirichenko committed:
Update style.css

Alexander-Kirichenko committed:
Update style.css

Alexander-Kirichenko committed:
Update style.css

Alexander-Kirichenko committed:
делает декоративные элементы на главной странице

Alexander-Kirichenko committed:
сделал список преимуществ с pixelPerfect

Alexander-Kirichenko committed:
Update style.css

Alexander-Kirichenko committed:
работает над декоративными элементами

Alexander-Kirichenko committed:
работает над декоративными элементами

Alexander-Kirichenko committed:
сделал favorite

Alexander-Kirichenko committed:
работает над декоративными элементами

Alexander-Kirichenko committed:
работает над декоративными элементами

Alexander-Kirichenko committed:
завершает (?) декорацию футера

Alexander-Kirichenko committed:
завершает 90% главной страницы

Alexander-Kirichenko committed:
продолжает стилизацию

Alexander-Kirichenko committed:
продолжает декарацию

Alexander-Kirichenko committed:
декорирует каатлог

Alexander-Kirichenko committed:
стилизует каталог

Alexander-Kirichenko committed:
стилизует каталог

Alexander-Kirichenko committed:
работает над состоянием кнопок

Alexander-Kirichenko committed:
состояния пагинации

Alexander-Kirichenko committed:
кнопка подписки

Alexander-Kirichenko committed:
сделал рейтинг

Alexander-Kirichenko committed:
рейтинг

Alexander-Kirichenko committed:
Update style.css

Alexander-Kirichenko committed:
ошибки валидатора

Alexander-Kirichenko committed:
поменял некоторые border на outline

Alexander-Kirichenko committed:
Update style.css

pavel-niukalo commented on catalog.html:
```diff
@@ -185,6 +208,12 @@ <h3>Amara Resort & Spa</h3>
                 <button type="button" class="button button-favorite">
                   в избранное
                 </button>
+                <div class="stars four-stars">
+                  <img src="images/catalog/star.svg" alt="">
+                  <img src="images/catalog/star.svg" alt="">
+                  <img src="images/catalog/star.svg" alt="">
+                  <img src="images/catalog/star.svg" alt="">
+                </div>
```
Насчет звезд. Как бы я предложил сделать.
Положить их все-таки фоном, чтобы не засорять разметку, и дать им `background-repeat: space`. А уже дополнительными классами менять ширину, чтобы показывалось разное кол-во звезд

pavel-niukalo commented on styles/style.css:
```diff
@@ -120,13 +122,36 @@ body {
   text-transform: capitalize;
 }
 
+.navigation-link.current:after {
+  position: absolute;
+  width: 68px;
+  right: 20px;
+  bottom: 0;
+  left: 16px;
+  height: 2px;
+  background-color: #756257;
+  content: "";
+}
+
+.navigation-link.hotels.current:after {
+  position: absolute;
+  width: 93px;
+  right: 20px;
+  bottom: 0;
+  left: 16px;
+  height: 2px;
+  background-color: #756257;
+  content: "";
+}
+
```
Насчет этой черточки. Получается не очень универсально. Во-первых, для каждой ссылки придется делать доп. класс. Во-вторых, представь если поменяется текст хотя бы на букву. Придется переделывать

Я бы предложил сделать следующим образом:
- даем ширину в 100% от ссылки
- но у ссылки есть паддинги, а нам надо чтобы только по тексту все было. В таком случае делаем `width: calc(100% - 32px)` 32 - это 16 слева и 16 справа

![image](https://user-images.githubusercontent.com/52506235/221420619-30644ede-d131-4023-8a4d-dc577c4f0d60.png)

Alexander-Kirichenko replied:
ширину выставил, как ты сказал, всё работает, даже на длинных текстах. Но не понял про ```left``` и ```transform```. у меня эти свойства не работают, работают именно что обычные left-right

pavel-niukalo replied:
`transform` смещает элемент и это такой хак небольшой. Сначала мы сместили элемент в середину родительского (`50%`), теперь он начинается от середины родителя. А потом за счет `transform: translateX(-50%)` смещаем элемент влево на 50% уже его ширины. И таким образом мы отцентрируем

pavel-niukalo commented on styles/style.css:
```diff
@@ -199,26 +262,41 @@ body {
 }
 
+.advantages-item:not(.more-advantages-item) {
+  background-image: url("/images/index/rec.svg");
```
Тут нужно `../images`, иначе на опубликованной версии не отображается

![image](https://user-images.githubusercontent.com/52506235/221420778-5f602990-ed68-4b3c-9584-590e6025b4e4.png)

Alexander-Kirichenko replied:
исправил пути, все 4. нам ведь всегда надо ```../``` делать, это мы ВЫХОДИМ из папки стилей, и только потом заходим в img, так?

pavel-niukalo replied:
Да, все верно

pavel-niukalo commented on styles/style.css:
```diff
@@ -199,26 +262,41 @@ body {
+
+.advantages-item-town div,
+.advantages-item-bridge div {
+  background-image: url("/images/index/rec.png");
```
+1

pavel-niukalo commented on styles/style.css:
```diff
@@ -475,14 +601,115 @@ body {
+  fill: rgba(255, 255, 255, 0.3);
+}
+
+.bread-crumbs li:last-child::before {
+  content: "";
+  border: solid white;
+  border-width: 0 2px 2px 0;
+  display: inline-block;
+  padding: 3px;
+  transform: translateY(-1px) rotate(-45deg);
+  margin: 0 9px 0 7px;
+}
```
Тут нужно поработать над переполнением. Если добавится элементов, то во-первых, между ними нет отступа, во-вторых стрелочка отображается только у последнего, а по идее надо чтобы между каждым было

![image](https://user-images.githubusercontent.com/52506235/221420951-267a338b-319e-4366-ae98-ce8ba6840a11.png)

Я бы пошел от обратного, то есть добавлял бы не последнему, а каждому, кроме последнего

pavel-niukalo commented on styles/style.css:
```diff
@@ -475,14 +601,115 @@ body {
+}
+
+.checkbox-input:checked + .checkbox-custom {
+  background-image: url("/images/catalog/checked.svg");
```
+1 про путь

pavel-niukalo commented on styles/style.css:
```diff
@@ -475,14 +601,115 @@ body {
+}
+
+.radio-input:checked + .radio-custom {
+  background-image: url("/images/catalog/radio.svg");
```
+1

pavel-niukalo commented on styles/style.css:
```diff
@@ -475,14 +601,115 @@ body {
+  opacity: 0.5;
+}
+
+.radio-input:focus + .radio-custom {
```
Еще нужно добавить состояние `disabled` для чекбоксов и радио, к этому часто придираются

pavel-niukalo commented on styles/style.css:
```diff
@@ -560,18 +839,81 @@ body {
+.catalog-sorting-select [aria-label] {
+  padding: 14px 20px;
+}
+
+
```
Лишняя пустая строка добавилась

Alexander-Kirichenko replied:
убрал

Alexander-Kirichenko committed:
исправляет ширину и убирает ненужный класс в навигации

Alexander-Kirichenko committed:
исправляет пути к img

Alexander-Kirichenko committed:
работает над зывёздами

Alexander-Kirichenko committed:
формы

Alexander-Kirichenko committed:
стилизует

Alexander-Kirichenko committed:
модальное окно

Alexander-Kirichenko committed:
модальное окно

Alexander-Kirichenko committed:
делает модальное окно

Alexander-Kirichenko committed:
да

Alexander-Kirichenko committed:
модальное окно

Alexander-Kirichenko committed:
Update style.css

pavel-niukalo commented on catalog.html:
```diff
@@ -185,6 +234,8 @@ <h3>Amara Resort & Spa</h3>
                 <button type="button" class="button button-favorite">
                   в избранное
                 </button>
+                <div class="stars four-stars">
+                </div>
```
Уже хорошо, но нужно еще вернуть подпись для скринридеров скрытую. По типу "Четыре звезды." и тд

pavel-niukalo commented on index.html:
```diff
@@ -191,51 +223,99 @@ <h3 class="visually-hidden">Седона в социальных сетях</h3>
-            <img src="images/index/modal-date.svg" width="16" height="15" alt="Иконка даты.">
-          </fieldset>
+            <input class="input-date-in" id="modal-date-in" type="text" name="date-in" placeholder="27 апреля 2020">
+            <button class="calendar-icon" type="button"></button>
```
В эти кнопочки тоже нужно добавить скрытую подпись, которая скажет, что произойдет если пользователь нажмет

pavel-niukalo commented on index.html:
```diff
@@ -191,51 +223,99 @@ <h3 class="visually-hidden">Седона в социальных сетях</h3>
+            <label class="kids" for="amount-kids">
+              <span class="kids-text">Дети:</span>
+
+              <button class="pop-up-button">
```
Так же не забывай указывать `type`

Alexander-Kirichenko replied:
указал type=button

pavel-niukalo commented on index.html:
```diff
@@ -191,51 +223,99 @@ <h3 class="visually-hidden">Седона в социальных сетях</h3>
-      </div>
-      <button type="button" class="button button-close"><span class="visually-hidden">Закрыть.</span></button>
-    </div> -->
+          <button class="button button-find" type="submit">
+            <span class="button-text">Найти</span>
+          </button>
```
Точно ли нужен `span`?

Alexander-Kirichenko replied:
не нужен

pavel-niukalo commented on styles/style.css:
```diff
@@ -638,12 +1091,81 @@ body {
+  background-color: #7DB54F;
+}
+
+.stars {
+  background-image: url("../images/catalog/star.svg");
+  background-repeat: space;
+  align-self: center;
+  width: 140px;
+  height: 17px;
+}
+
+.four-stars {
+  background-image: url("../images/catalog/star.svg");
+  background-repeat: space;
+  align-self: center;
+  width: 89px;
+  height: 17px;
+}
+
+.three-stars {
+  background-image: url("../images/catalog/star.svg");
+  background-repeat: space;
+  align-self: center;
+  width: 66px;
+  height: 17px;
+}
+
+.two-starts {
+  background-image: url("../images/catalog/star.svg");
+  background-repeat: space;
+  align-self: center;
+  width: 43px;
+  height: 17px;
+}
```
Дублировать свойства нет смысла, поскольку они унаследуются от `stars`. А вот доп. классами будем менять только ширину

```css
.stars {
  background-image: url("../images/catalog/star.svg");
  background-repeat: space;
  width: 140px;
  height: 17px;
}

.four-stars {
  width: 89px;
}

.three-stars {
  width: 66px;
}

.two-starts {
  width: 43px;
}
```

Смотри как код сократился

pavel-niukalo commented on styles/style.css:
```diff
@@ -681,7 +1218,360 @@ body {
+}
+
+.modal-window {
+  font-family: 'PT Sans';
```
Не забываем указывать семейство шрифтов

pavel-niukalo commented on styles/style.css:
```diff
@@ -681,7 +1218,360 @@ body {
+}
+
+.input-date-out {
+  font-family: 'PT Sans';
```
+1

pavel-niukalo commented on styles/style.css:
```diff
@@ -681,7 +1218,360 @@ body {
+  border-radius: 10px;
+  padding: 15px 18px 18px 22px;
+  box-shadow: 0 15px 30px rgba(0, 0, 0, 0.3);
+  font-family: 'PT Sans';
```
+1. В целом надо поглядеть, шрифт наследуюется же

pavel-niukalo commented on styles/style.css:
```diff
@@ -681,7 +1218,360 @@ body {
+  border-radius: 50%;
+  right: 52px;
+  top: 65px;
+  background-image: url(../images/index/modal-cross.svg);
```
Кавычки в `url`

Alexander-Kirichenko committed:
убрал спан у кнопки

забыл, что можно стилизовать прозрачность текста через rgba

Alexander-Kirichenko committed:
указывает type="button"

Alexander-Kirichenko committed:
убирает ненужные свойства у .star

Alexander-Kirichenko committed:
исправляет орфографию

Alexander-Kirichenko committed:
добавляет span для скринридеров

Alexander-Kirichenko committed:
добавляет кавычки

Alexander-Kirichenko committed:
модальное окно

- меняет Placeholder
- добавляет иконку для поп-апа
- работает над интерактивными элементами

Alexander-Kirichenko committed:
модальное окно, интерактивные элементы

Alexander-Kirichenko committed:
исправляет семейство шрифтов

не понял про наследование. убрал с дочерних, и на некоторых появился Arial

Alexander-Kirichenko committed:
выравнивает элементы для pixelperfect

Alexander-Kirichenko committed:
делает поп-ап поп-апом)

Alexander-Kirichenko committed:
ра

Alexander-Kirichenko committed:
min-height для модального окна

Alexander-Kirichenko committed:
Update index.html

Alexander-Kirichenko committed:
Update index.html

Alexander-Kirichenko committed:
Update style.css

Alexander-Kirichenko committed:
убирает высоту у img и делает object-fit

Alexander-Kirichenko committed:
тул-тип

Alexander-Kirichenko committed:
Update index.html

Alexander-Kirichenko committed:
Update index.html

Alexander-Kirichenko committed:
Update index.html

Alexander-Kirichenko committed:
дорабатывает модальное окно

Alexander-Kirichenko committed:
работает над состоянием disabled для критериев

Alexander-Kirichenko committed:
работает над состоянием disabled

Alexander-Kirichenko committed:
завершает работу над состояниями

Alexander-Kirichenko committed:
меняет pointer у disabled

Alexander-Kirichenko committed:
Update style.css

Alexander-Kirichenko committed:
завершил активные  элементы в навигации

Alexander-Kirichenko committed:
рефакторинг range-slider

Alexander-Kirichenko committed:
рефакторинг range-input

Alexander-Kirichenko committed:
завершает активные состояния для Input-range

Alexander-Kirichenko committed:
добавляет доступности кнопкам + -

Alexander-Kirichenko committed:
уементшает код в модальных инпутах

Alexander-Kirichenko committed:
до

Alexander-Kirichenko committed:
Update style.css

Alexander-Kirichenko committed:
проверил на переполнение чеклист

Alexander-Kirichenko committed:
делает папку images в соответствии с требованиями

Alexander-Kirichenko committed:
папка images в соответсвии с критериями

Alexander-Kirichenko committed:
рефакторинг иконки календаря

Alexander-Kirichenko committed:
критерий на переполнение

Alexander-Kirichenko committed:
Update style.css

Alexander-Kirichenko committed:
Update catalog.html

Alexander-Kirichenko committed:
Update index.html

Alexander-Kirichenko committed:
Update catalog.html

Alexander-Kirichenko committed:
Update catalog.html

Alexander-Kirichenko committed:
Update catalog.html

Alexander-Kirichenko committed:
Update index.html

Alexander-Kirichenko committed:
описание графики в Index

Alexander-Kirichenko committed:
role='img' для svg
